packages/app-store/_utils/useAddAppMutation.ts

```diff
@@ -42,17 +56,20 @@ function useAddAppMutation(_type: App["type"] | null, options?: Parameters<typeo
 
     if (!isOmniInstall) {
       window.location.href = json.url;
-      return;
-    }
-
-    // Skip redirection only if it is an OmniInstall and redirect URL isn't of some other origin
-    // This allows installation of apps like Stripe to still redirect to their authentication pages.
+    } else {
+      // Skip redirection only if it is an OmniInstall and redirect URL isn't of some other origin
+      // This allows installation of apps like Stripe to still redirect to their authentication pages.
 
-    // Check first that the URL is absolute, then check that it is of different origin from the current.
-    if (/https?:\/\//.test(json.url) && !json.url.startsWith(window.location.origin)) {
-      // TODO: For Omni installation to authenticate and come back to the page where installation was initiated, some changes need to be done in all apps' add callbacks
-      window.location.href = json.url;
+      // Check first that the URL is absolute, then check that it is of different origin from the current.
+      if (/https?:\/\//.test(json.url) && !json.url.startsWith(window.location.origin)) {
+        // TODO: For Omni installation to authenticate and come back to the page where installation was initiated, some changes need to be done in all apps' add callbacks
+        window.location.href = json.url;
+      }
     }
+
+    return {
+      setupRequiredForInstallation: json.url.includes("setupRequiredForInstallation=true"),
```

Comment:
Actually, this kind of solution will be needed in the case we need OAuth2 authorization by the user, redirecting the user to an external site, not just an internal setup page.